src/main/scala/monitors/TLMonitorInjToDcache.scala

```diff
@@ -0,0 +1,89 @@
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.monitors
+
+import chisel3.aop.injecting._
+
+import chisel3.experimental.{BaseModule, FixedPoint}
+import chisel3.{RawModule, VecInit}
+
+import chisel3.aop.injecting.InjectingAspect
+import chisel3.aop._
+import chisel3._
+import firrtl.options.TargetDirAnnotation
+import freechips.rocketchip.config.Parameters
+import freechips.rocketchip.config._
+import freechips.rocketchip.rocket._
+import freechips.rocketchip.stage.{ConfigsAnnotation, TopModuleAnnotation}
+//import freechips.rocketchip.system.{RocketChipStage, TestHarness}
+import freechips.rocketchip.system._
+import chisel3.internal.firrtl._
+//import freechips.rocketchip.util.HasRocketChipStageUtils
+import freechips.rocketchip.util.SelectDiplomacy
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.formal._
+import freechips.rocketchip.diplomacy._
+import scala.reflect.ClassTag
+
+/** API to access Diplomacy Node data:
+  *  {{{
+  *     val baseNodes = SelectDiplomacy.collectBaseNodes()
+  *     val impMod = SelectDiplomacy.collectImpModules()
+  *     val lazyMod = SelectDiplomacy.collectLazyModules()
+  * 
+  *     val mixedNodes = SelectDiplomacy.mixedNodes()
```

Comment:
whats a use case for all these? Why would i want to query diplomatic nodes like this? I can imagine wanting to grab for example all "TLSinkNodes" or "AXI4MasterNodes", but why would I abstractly want to query all `mixedNexusNode`?